### PR TITLE
[OR] Offset.fromDirection and Size.aspectRatio

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
Completely untested code right now. :-)

File: testing/dart/geometry_test.dart [hunk 1]
```diff
@@ -19,4 +20,29 @@ void main() {
     expect(const Offset(-1.0, 1.0).direction, pi * 3.0 / 4.0);
     expect(const Offset(-1.0, -1.0).direction, -pi * 3.0 / 4.0);
   });
+  test('Offset.fromDirection', () {
+    expect(new Offset.fromDirection(0.0, 0.0), const Offset(0.0, 0.0));
+    expect(new Offset.fromDirection(pi / 2.0), const Offset(0.0, 1.0));
+    expect(new Offset.fromDirection(-pi / 2.0), const Offset(0.0, -1.0));
+    expect(new Offset.fromDirection(0.0), const Offset(1.0, 0.0));
+    expect(new Offset.fromDirection(pi / 4.0), const Offset(1.0, 1.0));
```
[the remark this answers is not in the document]
Comment (Hixie) [reply testing/dart/geometry_test.dart]: the code was completely untested at the point where i pushed it, i wanted to get something on github so i could show a community member who was asking for it.

File: lib/ui/geometry.dart [hunk 2]
```diff
@@ -121,6 +121,15 @@ class Offset extends OffsetBase {
   /// and the second sets [dy], the vertical component.
   const Offset(double dx, double dy) : super(dx, dy);
 
+  /// Creates an offset from its [direction] and [distance].
+  ///
+  /// The direction is in radians clockwise from the positive x-axis.
+  ///
+  /// The distance can be omitted, to create a unit vector (distance = 1.0).
+  factory Offset.fromDirection(double direction, [ double distance = 1.0 ]) {
+    return new Offset(distance * math.cos(direction), - distance * math.sin(direction));
```
[the remark this answers is not in the document]
Comment (Hixie) [reply lib/ui/geometry.dart]: Yeah but this file hasn't been converted yet.

File: lib/ui/geometry.dart [hunk 3]
```diff
@@ -373,6 +382,23 @@ class Size extends OffsetBase {
...
+  ///
+  /// The sign of the aspect ratio depends on the sign of [width] and [height].
+  ///
+  /// See also:
+  ///
+  ///  * [AspectRatio], a widget for giving a child widget a specific aspect
+  ///    ratio.
+  ///  * [FittedBox], a widget that (in most modes) attempts to maintain a
+  ///    child widget's aspect ratio while changing its size.
+  double get aspectRatio => dx / dy;
```
[the remark this answers is not in the document]
Comment (Hixie) [reply lib/ui/geometry.dart]: oops, fixed.

**Comment by Hixie** [comment]
Ok, this time it's actually been tested.

**Comment by slightfoot** [comment]
@Hixie thanks 👍

**Comment by Hixie** [comment]
Rebased and kicked bots, will land on green.

**Pull request merged**